lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster_facts.py

```diff
@@ -0,0 +1,313 @@
+                            returned: always
+                            type: str
+                            sample: myuser
+        connectivity_endpoints:
+            description:
+                - Cluster's connectivity endpoints.
+            type: list
+            contains:
+                location:
+                    description:
+                        - Endpoint location.
+                    returned: always
+                    type: str
+                    sample: myCluster-ssh.azurehdinsight.net
+                name:
+                    description:
+                        - Endpoint name.
+                    returned: always
+                    type: str
+                    sample: SSH
+                port:
+                    description:
+                        - Endpoint port.
+                    returned: always
+                    type: int
+                    sample: 22
+                protocol:
+                    description:
+                        - Endpoint protocol.
+                    returned: always
+                    type: str
+                    sample: TCP
+        tags:
+            description:
+                - Tags
+            returned: always
+            type: complex
+            sample: {}
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+from ansible.module_utils.common.dict_transformations import _camel_to_snake
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.hdinsight import HDInsightManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMHDInsightclusterFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str'
+            ),
+            name=dict(
+                type='str'
+            ),
+            tags=dict(
+                type='list'
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False
+        )
+        self.mgmt_client = None
+        self.resource_group = None
+        self.name = None
+        self.tags = None
+
+        super(AzureRMHDInsightclusterFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(HDInsightManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        if self.name is not None:
+            self.results['clusters'] = self.get()
+        elif self.resource_group is not None:
+            self.results['clusters'] = self.list_all()
+        return self.results
```

Comment:
what if else? return hdinsight in current subs?

Reply:
seems by resource group you should be list_by_resource_group() else list_all()

Reply:
ok, fixed now